### PR TITLE
feat(AAE-896): Cloud Native Bpmn message events with stateful backend support

### PR DESCRIPTION
**Pull request by igdianov**
This PR extracted Message Service into dedicated repository and implements Message Events Producers and Message Command Consumer Api to be able to communicate with Messages Connector Service via Spring Cloud Stream. 

image

Depends on https://github.com/Activiti/activiti-api/pull/156

Test Coverage:

- [x] Intermediate Throw Message Event Producer
 
image

- [x] Intermediate Catch Message Event Producer

image

- [x] Start Message Event Command Executor

image

- [x] Boundary Task Message Event Producer

image

- [x] End Message Event Producer

image

- [x] Event Gateway Message Event Producer

image

- [x] Event Subprocess Start Message Event Producer

image

- [x] Boundary Subprocess Message Event Producer

image

- [x] Event Subprocess Start Message Event Non Interrupting Producer

image

- [x] Boundary Subprocess Message Event Non-Interrupting Producer

image

- [x] Boundary Task Message Event Non-Interrupting Producer

image

Part of https://github.com/Activiti/Activiti/issues/3028

### CODE REVIEW

File: activiti-cloud-services-runtime-bundle/messages-events/src/main/java/org/activiti/cloud/services/messages/events/producer/StartMessageDeployedEventMessageProducer.java [hunk 1]
```diff
@@ -35,9 +36,20 @@ public void onEvent(StartMessageDeployedEvent event) {
...
-                                                                               .build();
+        StartMessageSubscription messageSubscription = event.getEntity()
+                                                            .getMessageSubscription();
+
+        MessageEventPayload messageEventPayload = MessageEventPayloadBuilder.messageEvent(messageSubscription.getEventName())
+                                                                            .withCorrelationKey(messageSubscription.getConfiguration())
+                                                                            .build();
+        
+        
+        Message<MessageEventPayload> message = messageBuilderFactory.create(event)
```
Comment (erdemedeiros) [inline activiti-cloud-services-runtime-bundle/messages-events/src/main/java/org/activiti/cloud/services/messages/events/producer/StartMessageDeployedEventMessageProducer.java:41]: The content of the different message producer is very similar, especially `BpmnMessageReceivedEventMessageProducer`, `BpmnMessageSentEventMessageProducer` and `BpmnMessageWaitingEventMessageProducer`. Would it make sense to extract common code into a separated class?

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
@igdianov, I have only one comment concerning MessageProducers code. LTGM otherwise.

File: activiti-cloud-services-runtime-bundle/messages-events/src/main/java/org/activiti/cloud/services/messages/events/producer/StartMessageDeployedEventMessageProducer.java (hunk 1 above)
Comment (igdianov) [reply activiti-cloud-services-runtime-bundle/messages-events/src/main/java/org/activiti/cloud/services/messages/events/producer/StartMessageDeployedEventMessageProducer.java:41]: I think it's a good practice to use explicit bpmn listeners to consume engine message events. But I agree that the repeating  boilerplate code could be refactored.

Comment (igdianov) [reply activiti-cloud-services-runtime-bundle/messages-events/src/main/java/org/activiti/cloud/services/messages/events/producer/StartMessageDeployedEventMessageProducer.java:41]: Fixed in 08ab8b8953897644f446f0eb5d6e759115bfc43f

**Review by erdemedeiros — APPROVED** [review]

**Pull request merged**